controllers/csi/gc/logs.go

```diff
@@ -0,0 +1,100 @@
+package csigc
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/afero"
+)
+
+const (
+	maxLogFolderSizeBytes = 300000
+	maxNumberOfLogFiles   = 1000
+	maxLogAge             = 14 * 24 * time.Hour
+)
+
+type logFileInfo struct {
+	UnusedVolumeIDs []os.FileInfo
+	NumberOfFiles   int64
+	OverallSize     int64
+}
+
+func (gc *CSIGarbageCollector) runLogGarbageCollection(tenantUUID string) {
+	fs := &afero.Afero{Fs: gc.fs}
+	logs, err := gc.getLogFileInfo(tenantUUID, fs)
+	if err != nil {
+		gc.logger.Info("failed to get log file information")
+		return
+	}
+
+	shouldDelete := logs.NumberOfFiles > 0 && (logs.OverallSize > maxLogFolderSizeBytes || logs.NumberOfFiles > maxNumberOfLogFiles)
+	if shouldDelete {
+		gc.tryRemoveLogFolders(logs.UnusedVolumeIDs, tenantUUID, fs)
+	}
```

Comment:
This condition should have test coverage.

Reply:
Move to a function and tested